src/contracts/wrappers/ERC1155FloorFactory.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
+
+import {IERC1155FloorFactory} from "../interfaces/IERC1155FloorFactory.sol";
+import {ERC1155FloorWrapper} from "../wrappers/ERC1155FloorWrapper.sol";
+import {WrapperErrors} from "../utils/WrapperErrors.sol";
+import {Ownable} from "../utils/Ownable.sol";
+import {Proxy} from "../utils/Proxy.sol";
+import {IDelegatedERC1155Metadata, IERC1155Metadata} from "../interfaces/IDelegatedERC1155Metadata.sol";
+
+contract ERC1155FloorFactory is IERC1155FloorFactory, Ownable, IDelegatedERC1155Metadata, WrapperErrors {
+    mapping(address => address) public override tokenToWrapper;
+    ERC1155FloorWrapper private wrapperImpl;
+
+    IERC1155Metadata internal metadataContract; // address of the ERC-1155 Metadata contract
+
+    constructor(address _admin) Ownable(_admin) {
+        wrapperImpl = new ERC1155FloorWrapper();
+        wrapperImpl.initialize(address(0));
+    }
+
+    /**
+     * Creates an ERC-1155 Floor Wrapper for given token contract
+     * @param tokenAddr The address of the ERC-1155 token contract
+     * @return wrapperAddr The address of the ERC-1155 Floor Wrapper
+     */
+    function createWrapper(address tokenAddr) external returns (address wrapperAddr) {
+        if (tokenToWrapper[tokenAddr] != address(0)) {
+            revert WrapperAlreadyCreated(tokenAddr, tokenToWrapper[tokenAddr]);
+        }
+
+        // Compute the address of the proxy contract using create2
+        bytes memory code = abi.encodePacked(type(Proxy).creationCode, uint256(uint160(address(wrapperImpl))));
+        bytes32 salt = keccak256(abi.encodePacked(tokenAddr));
+
+        // Deploy it
+        assembly {
+            wrapperAddr := create2(0, add(code, 32), mload(code), salt)
+        }
+        if (wrapperAddr == address(0)) {
+            revert WrapperCreationFailed(tokenAddr);
+        }
+        ERC1155FloorWrapper(wrapperAddr).initialize(tokenAddr);
+
+        tokenToWrapper[tokenAddr] = wrapperAddr;
```

Comment:
You can skip this mapping and implement `tokenToWrapper` by recomputing the address (and then checking if the address has any code).